### DISCUSSION TITLE
Github Classroom Autograding - How to test Junit tests?

### DISCUSSION THREAD

**Discussion by LuisDamas** [community/community#70690] [GitHub Education]
### Hi everyone,

How can we test if a JUnit test is well implemented in Github classroom? Suppose the student is required to write the function int triple(int n) (using java) and also to implement a test to check if the implementation is correct. How can we check if the student's unit test void tripleFunctionIsOK() method is doing its job correctly?
The goal is to check if the student can write proper unit tests.

**Comment by uhafner** [comment]
I am using mutation testing to see if the tests are well written (Java projects with PIT mutation testing). You can also use code coverage tools to see if at least the code has been used in the tests.
